apps/charterafrica/src/lib/ecosystem/payload.js

```diff
@@ -0,0 +1,75 @@
+import * as Sentry from "@sentry/nextjs";
+
+import localize from "@/charterafrica/lib/ecosystem/utils";
+import api from "@/charterafrica/lib/payload";
+
+export async function updateOrCreate(collection, toCreate, locale) {
+  try {
+    const { docs } = await api.getCollection(collection, {
+      locale,
+      where: {
+        airtableId: { equals: toCreate.airtableId },
+      },
+    });
+    if (docs.length) {
+      const data = await api.updateCollection(
+        collection,
+        docs[0]?.id,
+        toCreate,
+        { locale }
+      );
+      return data;
+    }
+    const data = await api.createCollection(collection, toCreate, { locale });
+    return data;
+  } catch (e) {
+    Sentry.captureMessage(e.message);
+    return null;
+  }
+}
+
+export async function createCollection(collection, toCreate, { localized }) {
+  try {
+    const locales = localized ? ["en", "pt", "fr"] : ["en"];
+    const localizedData = localize(toCreate || {}, locales);
+    if (!localizedData) {
+      return null;
+    }
+    if (!localized) {
+      return updateOrCreate(collection, localizedData?.en);
+    }
+    const promises = Object.keys(localizedData).map((key) =>
+      updateOrCreate(collection, localizedData?.[key], key)
+    );
+    return Promise.all(promises);
+  } catch (e) {
+    Sentry.captureMessage(e.message);
+    return null;
+  }
+}
+
+export async function bulkMarkDeleted(collection, fromSource) {
+  try {
+    const dataIds = fromSource.map((item) => item?.airtableId).join(",");
+    const { docs: toDelete } = await api.getCollection(collection, {
+      where: {
+        airtableId: {
+          not_in: dataIds,
+        },
+      },
+    });
+    Promise.all(
+      toDelete.map(async ({ id }) => {
+        try {
+          await api.updateCollection(collection, id, {
+            deletedAt: new Date(),
+          });
+        } catch (error) {
+          Sentry.captureMessage(error.message);
+        }
+      })
+    );
```

Comment:
```suggestion
    Promise.all(
      toDelete.map(async ({ id }) => {
        try {
          await api.updateCollection(collection, id, {
            deletedAt: new Date(),
          });
        } catch (error) {
          Sentry. captureException(error.message);
        }
      })
    );
```

If we're catching the error and sending it to Sentry (which is correct), how do we tell the promise to reject?

Reply:
@kilemensi do we need to terminate the whole process because a deletion failed?

Thinking we can only skip if an error occurred during deletion and have it reported in sentry

Reply:
1. Aren't we going to end up with wrong data?
2. If we need to skip, then `Promise.allsettled` is what we should be using, no?

Reply:
Perhaps promise.allSettled can work here

Reply:
But what are we going to do about the failed deletes?